### PR TITLE
feat: unskip 65 more activesupport tests (time-zone, time-travel, class-attribute)

### PR DESCRIPTION
**Pull request by deanmarano**
## Summary

Continuing the activesupport test unskip work. This batch focuses on the TimeZone class and a few smaller files.

**time-zone (+42 tests):** Added tomorrow(), yesterday(), iso8601(), rfc3339(), match(), and usZones() methods to TimeZone. Implemented tests for ISO 8601/RFC 3339 parsing with fractional seconds, zones, nil, old dates, far future dates, DST jumps, UTC offset behavior, zone matching, and more. The remaining 31 skips are mostly strptime (no JS equivalent), country zones, YAML, and some DST blackout edge cases.

**time-travel (+9 tests):** Implemented usec-related tests and MySQL rounding test with ms assertions.

**class/attribute (+14 tests):** Implemented core class_attribute behavior tests. Remaining 5 skips need Ruby-specific features (singleton classes, prepend).

**Results:**
- time_zone: 35/73 skipped -> 77/31 skipped
- time_travel: 14/13 skipped -> 23/4 skipped  
- class/attribute: 0/19 skipped -> 14/5 skipped
- Overall activesupport: 67.6% -> 69.1%

### CODE REVIEW

File: packages/activesupport/src/time-zone.test.ts [hunk 1]
```diff
@@ -229,15 +313,52 @@ describe("TimeZoneTest", () => {
...
+
+  it("parse far future date with time zone offset in string", () => {
+    const zone = TimeZone.find("Eastern Time (US & Canada)");
+    const twz = zone.parse("2050-01-01T00:00:00-05:00");
+    expect(twz.utc().getUTCFullYear()).toBe(2050);
+  });
+
+  it("parse returns nil when string without date information is passed in", () => {
+    const zone = TimeZone.find("Eastern Time (US & Canada)");
+    expect(() => zone.parse("foobar")).toThrow();
```
[the remark this answers is not in the document]
Comment (deanmarano) [reply packages/activesupport/src/time-zone.test.ts]: In Rails, parse returns nil for unparseable strings. In our TS version, we throw instead since returning null would require a TimeWithZone | null return type throughout the API. Throwing is the idiomatic TS equivalent — callers can catch if needed. The test name can't change per project rules.

File: packages/activesupport/src/time-zone.ts [hunk 2]
```diff
@@ -470,6 +470,83 @@ export class TimeZone {
...
+   */
+  iso8601(str: string | null | undefined): TimeWithZone {
+    if (str == null || str.trim() === "") {
+      throw new Error("invalid date");
+    }
+    const trimmed = str.trim();
+    if (!/^\d{4}-?\d{2}-?\d{2}(T\d{2}:?\d{2}(:?\d{2})?)?/.test(trimmed)) {
+      throw new Error("invalid date");
+    }
+    return this.parse(trimmed);
```
[the remark this answers is not in the document]
Comment (deanmarano) [reply packages/activesupport/src/time-zone.ts]: Fixed — added end-anchored ISO 8601 format validation with optional fractional seconds and timezone offsets.

File: packages/activesupport/src/time-zone.ts [hunk 3]
```diff
@@ -470,6 +470,83 @@ export class TimeZone {
...
+  }
+
+  /**
+   * Parse an RFC 3339 string in this timezone.
+   */
+  rfc3339(str: string): TimeWithZone {
+    const trimmed = str?.trim() ?? "";
+    if (!/^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}([.]\d+)?(Z|[+-]\d{2}:?\d{2})$/.test(trimmed)) {
```
[the remark this answers is not in the document]
Comment (deanmarano) [reply packages/activesupport/src/time-zone.ts]: Fixed — end-anchored regex, requires colon in offset per RFC 3339, trims input.
